controllers/druid/handler.go

```diff
@@ -592,49 +533,45 @@ func execCheckCrashStatus(sdk client.Client, nodeSpec *v1alpha1.DruidNodeSpec, m
 	}
 }
 
-func checkCrashStatus(sdk client.Client, m *v1alpha1.Druid) {
-	podList := makePodList()
-	listOpts := []client.ListOption{
-		client.InNamespace(m.Namespace),
-		client.MatchingLabels(makeLabelsForDruid(m.Name)),
-	}
-
-	if err := sdk.List(context.TODO(), podList, listOpts...); err != nil {
-		e := fmt.Errorf("failed to list pods for [%s:%s] due to [%s]", m.Kind, m.Name, err.Error())
-		sendEvent(sdk, m, v1.EventTypeWarning, "LIST_FAIL", e.Error())
-		logger.Error(e, e.Error(), "name", m.Name, "namespace", m.Namespace)
-	}
+func checkCrashStatus(sdk client.Client, drd *v1alpha1.Druid) error {
 
-	pod := make([]*v1.Pod, 0)
-	for i := range podList.Items {
-		pod = append(pod, &podList.Items[i])
+	podList, err := readers.List(context.TODO(), sdk, drd, makeLabelsForDruid(drd.Name), func() runtime.Object { return makePodList() }, func(listObj runtime.Object) []object {
+		items := listObj.(*v1.PodList).Items
+		result := make([]object, len(items))
+		for i := 0; i < len(items); i++ {
+			result[i] = &items[i]
+		}
+		return result
```

Comment:
I see similar logic iterated for almost every kind of object statefulset, PVC, podlist? Any way this can be refactored?